setup.py

```diff
@@ -13,10 +13,18 @@
     print("Error: version of setuptools is too old (<42)!")
     sys.exit(1)
 
+def get_cupy() -> str:
+    CUDA_VERSION = os.environ.get("CUDA_VERSION", None)
+    if CUDA_VERSION>="11.2": # after 11.2 use
+        cupy_pack = f"cupy-cuda{CUDA_VERSION[:2]}x"
+    else:
+        cupy_pack = f"cupy-cuda{CUDA_VERSION[:4].replace('.','')}"
+    return cupy_pack
```

Comment:
This fails if the env variable is not set. Can we have a check that in this case does not install any cupy or installs just the latest ? 
Also you could do a fall through check: env variable -> nvcc subprocess -> nvidia-smi.

Reply:
I considered the approach of nvcc and nvidia-smi, and there will encounter some issues like
1. no env var set
2. nvcc not available in the devel image, like `nvidia/cuda:11.7.1-runtime-ubuntu22.04`
3. nvidia-smi always shows the latest CUDA version 12.2 by the host driver while docker is using 11.7
or even has no nvidia-smi for some cases I have seen.

Can I list all cuda versions in cuda directory `/usr/local/cuda*` ?
while I have no idea about windows cuda

Reply:
Yeah as usual windows is the annoying part. This is why I though you could do a fall through approach, trying one after the other and if nothing works just assume a version ?